compiler/rustc_mir_transform/src/known_panics_lint.rs

```diff
@@ -102,8 +102,9 @@ impl<'tcx> Value<'tcx> {
                 }
                 (PlaceElem::Index(idx), Value::Aggregate { fields, .. }) => {
                     let idx = prop.get_const(idx.into())?.immediate()?;
-                    let idx = prop.ecx.read_target_usize(idx).ok()?;
-                    fields.get(FieldIdx::from_u32(idx.try_into().ok()?)).unwrap_or(&Value::Uninit)
+                    let idx = prop.ecx.read_target_usize(idx).ok()?.try_into().ok()?;
+                    let idx = (idx <= FieldIdx::MAX_AS_U32).then(|| FieldIdx::from_u32(idx))?;
+                    fields.get(idx).unwrap_or(&Value::Uninit)
```

Comment:
I don't think it's correct to `.unwrap_or(&Value::Uninit)` here. Why not just use `?` to early-return?

Reply:
Wait, that's preexisting :thinking: